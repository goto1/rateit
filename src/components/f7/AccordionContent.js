import React from "react";
import PropTypes from "prop-types";

const AccordionContent = ({ className, children }) => {
  const classNames = `accordion-item-content ${className
    ? className
    : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

AccordionContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default AccordionContent;
