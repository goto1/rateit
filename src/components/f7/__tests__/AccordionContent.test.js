import React from "react";
import AccordionContent from "../AccordionContent";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("AccordionContent component", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <AccordionContent>
          <span>HelloWorld!</span>
        </AccordionContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render a span", () => {
    const wrapper = shallow(
      <AccordionContent>
        <span>HelloWorld!</span>
      </AccordionContent>
    );
    const actual = wrapper.find("span").at(0).html();
    const expected = "<span>HelloWorld!</span>";
    expect(actual).toBe(expected);
  });

  it("should render with a specified className", () => {
    const wrapper = shallow(
      <AccordionContent className="custom">HelloWorld!</AccordionContent>
    );
    const actual = wrapper.props().className;
    const expected = "accordion-item-content custom";
    expect(actual).toBe(expected);
  });
});
