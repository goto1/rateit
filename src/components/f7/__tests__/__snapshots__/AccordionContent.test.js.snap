// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionContent component should render correctly 1`] = `


<div
  className="accordion-item-content"
>
  <span>
    HelloWorld!
  </span>
</div>
`;
